Core/include/Acts/Vertexing/ImpactPointEstimator.hpp

```diff
@@ -85,63 +92,88 @@ class ImpactPointEstimator {
-  /// at point of closest approach in 3d to given
-  /// reference position. The parameters and errors
-  /// are defined on the plane intersecting the track
-  /// at point of closest approach, with track orthogonal
-  /// to the plane and center of the plane defined as the
-  /// given reference point (vertex).
+  Result<double> calculateDistance(const GeometryContext& gctx,
+                                   const BoundTrackParameters& trkParams,
+                                   const Vector3& vtxPos, State& state) const;
+
+  /// @brief Estimates the track parameters at the 3D PCA (i.e., a point of
+  /// minimal 3D distance) to a vertex. The track parameters are defined wrt a
+  /// reference plane that has its origin at the vertex position and whose
+  /// z-axis points in the direction of the track momentum. The plane's x-axis
+  /// points approximately from the vertex to the 3D PCA (it is only approximate
+  /// because we force it to be orthogonal to the z-axis). The y-axis is
+  /// calculated as a cross product between x- and z-axis.
   ///
   /// @param gctx The geometry context
   /// @param mctx The magnetic field context
   /// @param trkParams Track parameters
   /// @param vtxPos Reference position (vertex)
   /// @param state The state object
   ///
-  /// @return New track params
+  /// @return Track parameters at the 3D PCA
   Result<BoundTrackParameters> estimate3DImpactParameters(
       const GeometryContext& gctx, const Acts::MagneticFieldContext& mctx,
       const BoundTrackParameters& trkParams, const Vector3& vtxPos,
       State& state) const;
 
-  /// @brief Estimates the compatibility of a
-  /// track to a vertex position based on the 3d
-  /// distance between the track and the vertex
+  /// @brief Estimates the compatibility of a track to a vertex based on their
+  /// 3D (if nDim = 3) or 4D (if nDim = 4) distance and the track covariance.
+  /// @note Confusingly, a *smaller* compatibility means that a track is *more*
+  /// compatible.
   ///
+  /// @tparam nDim Number of dimensions used to compute compatibility
+  /// @note If nDim = 3 we only consider spatial dimensions; if nDim = 4, we
+  /// also consider time. Other values are not allowed.
   /// @param gctx The Geometry context
   /// @param trkParams Track parameters at point of closest
-  /// approach in 3d as retrieved by estimate3DImpactParameters
+  /// approach in 3D as retrieved by estimate3DImpactParameters
   /// @param vertexPos The vertex position
   ///
   /// @return The compatibility value
-  Result<double> get3dVertexCompatibility(const GeometryContext& gctx,
-                                          const BoundTrackParameters* trkParams,
-                                          const Vector3& vertexPos) const;
-
-  /// @brief Estimates the impact parameters and their errors of a given
-  /// track w.r.t. a vertex by propagating the trajectory state
-  /// towards the vertex position.
+  template <unsigned int nDim = 3>
```

Comment:
I think I would prefer not defaulting this, so that at the call site it's a bit more obvious what's happening.